appside/nav/nav.js

```diff
@@ -103,16 +103,16 @@ class FrameFactory {
      */
     static build(frame) {
         // TODO: more sane identification of frame type (type property)
-        if(frame.hasOwnProperty('text')) {
+        if(frame.template === INTRO_FRAME_TEMPLATE) {
             return new IntroFrame(frame);
-        }
-        else if(frame.hasOwnProperty('statements')) {
+        } else if(frame.template === STATEMENTS_FRAME_TEMPLATE) {
             // TODO: identify when a different subtype of body frame is needed
             return new StatementsBodyFrame(frame);
-        }
-        else if(frame.hasOwnProperty('matched_emotions')) {
+        } else if(frame.template === SUMMARY_COUNT_FRAME_TEMPLATE) {
             // TODO: identify when this should be qualifier
```

Comment:
delete this comment

Reply:
removed all unnecessary comments from this file.